src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -34,6 +36,31 @@ private class CacheItem
             public TxOut[] OriginalOutputs;
         }
 
+        /// <summary>
+        /// Rewind data with its byte size that will be kept in the queue until it is persisted to a storage
+        /// </summary>
+        private class QueuedRewindData
+        {
+            /// <summary>Rewind data to be saved.</summary>
+            public RewindData RewindData { get; set; }
+
+            /// <summary>The size of the rewind data in bytes.</summary>
+            public long? DataSize { get; set; }
+
+            public static QueuedRewindData Create(RewindData rewindData)
```

Comment:
Why not normal constructor?

Reply:
Usually if I have a non-trivial constructor code (in this case serialization) I prefer using factory pattern as it gives you a much better flexibility. This is also consistent with MS guide: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/constructor and https://refactoring.guru/replace-constructor-with-factory-method. Adding a constructor might also force us to create a default constructor if serialization is needed. 

Having said all this, if we are getting away from serialization and byte size tracking, I have more than happy to move it to a constructor.

Reply:
Actually after reading what you linked, it looks like they give us plenty of arguments against this construction here, which seems unjustified. I don't really care that much, just looks odd to use it for such a trivial construction